### PR TITLE
Rust 1.41 fixes

### PR DESCRIPTION
**Pull request by jgrund**
Signed-off-by: Joe Grund <jgrund@whamcloud.io>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1546)

### CODE REVIEW

File: iml-agent/src/systemd.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ async fn wait_for_state(
 
 fn clean_bus_output(output: &str) -> Result<&str, ImlAgentError> {
     output
-        .split("\"")
+        .split('\"')
```
Comment (ip1981) [inline iml-agent/src/systemd.rs]: Does it really need a slash now?

File: iml-manager-cli/src/ostpool.rs [hunk 2]
```diff
@@ -94,69 +94,72 @@ async fn pool_lookup(fsname: &str, poolname: &str) -> Result<OstPoolApi, ImlMana
...
         }
+    };
+
+    let table = generate_table(&["Filesystem", "Pool Name", "OST Count"], xs);
+    table.printstd();
+
+    Ok(())
+}
+
+#[allow(clippy::cognitive_complexity)]
```
Comment (ip1981) [inline iml-manager-cli/src/ostpool.rs]: wow :)

File: iml-warp-drive/src/error.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ pub enum ImlWarpDriveError {
     ImlRabbitError(ImlRabbitError),
     ImlManagerClientError(ImlManagerClientError),
     TokioPostgresError(iml_postgres::Error),
-    DbError(DbError),
+    DbError(Box<DbError>),
```
Comment (ip1981) [inline iml-warp-drive/src/error.rs]: Why is this?

File: tests/integration/utils/chroma_log_collector.py [hunk 4]
```diff
@@ -180,9 +180,7 @@ def fetch_iml_diagnostics(self, server):
         errors.append(self.fetch_log(server, "/var/tmp/%s" % diagnostics, ""))
 
         if shell_run(["chmod", "777", "%s/%s" % (self.destination_path, diagnostics)]).rc:
-            errors.append(
-                "Unable to change perms on diagnostics at %s/%s" % (self.destination_path, diagnostics)
-            )
+            errors.append("Unable to change perms on diagnostics at %s/%s" % (self.destination_path, diagnostics))
```
Comment (ip1981) [inline tests/integration/utils/chroma_log_collector.py]: This is unrelated to Rust, no?

File: iml-warp-drive/src/error.rs (hunk 3 above)
Comment (nlinker) [reply iml-warp-drive/src/error.rs]: To mitigate `clippy::large_enum_variant`. One way is to wrap large variant into `Box` or just supress the warning.

Comment (jgrund) [reply iml-warp-drive/src/error.rs]: The enum will always be sized to it's largest variant. By putting the value in a box, the item is moved to the heap so it's just a pointer now.

Comment (ip1981) [reply iml-warp-drive/src/error.rs]: Ok, I got it :)

**Comment by jgrund** [comment]
___
*[tests/integration/utils/chroma_log_collector.py, line 183 at r3](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1546#-Lzs_l6hVDdYrjaU0UDF-r3-183:-LzvscVTABI9sL9xeyn9:by7lzmh) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/60ae33631d233d30c45ecf4cd5a6a33c3b051f31/tests/integration/utils/chroma_log_collector.py#L183)):*
<details><summary><i>Previously, ip1981 (Igor Pashev) wrote…</i></summary><blockquote>

This is unrelated to Rust, no?
</blockquote></details>

Yes, but python formatting was also broken, so may as well fix that in here too.

**Comment by jgrund** [comment]
___
*[iml-manager-cli/src/ostpool.rs, line 158 at r3](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1546#-LzuZ2ky6riKWJF95wpz-r3-158:-LzvzUUd72vm6vdgX0zQ:bamf9y4) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/60ae33631d233d30c45ecf4cd5a6a33c3b051f31/iml-manager-cli/src/ostpool.rs#L158)):*
<details><summary><i>Previously, mkpankov (Michael Pankov) wrote…</i></summary><blockquote>

This shouldn't be necessary after factoring out `ostpool_list`.

@jgrund would you try removing this going forward?
</blockquote></details>

Yeah, I'd rather have less `allow` than more.

**Comment by jgrund** [comment]
___
*[iml-agent/src/systemd.rs, line 37 at r3](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1546#-Lzs_l6gl8KX-U3wtXTt-r3-37:-LzvzqZTCGHuxzgiui9q:b-896fix) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/60ae33631d233d30c45ecf4cd5a6a33c3b051f31/iml-agent/src/systemd.rs#L37)):*
<details><summary><i>Previously, mkpankov (Michael Pankov) wrote…</i></summary><blockquote>

Agree, this looks incorrect
</blockquote></details>

Done.

File: iml-manager-cli/src/ostpool.rs [hunk 5]
```diff
@@ -94,69 +94,72 @@ async fn pool_lookup(fsname: &str, poolname: &str) -> Result<OstPoolApi, ImlMana
...
+                .filter_map(|p| {
+                    let id = extract_api_id(&p.ost.filesystem);
 
-                    let fs_ids = Vec::from_iter(fs_ids).join(",");
+                    id.map(|x| x.to_string())
+                })
+                .collect::<std::collections::HashSet<String>>();
 
-                    let mut query = Filesystem::query();
+            let fs_ids = Vec::from_iter(fs_ids).join(",");
```
Comment (ip1981) [inline iml-manager-cli/src/ostpool.rs]: nit: I don't like shadowing :)

Maybe it could be `... ..collect::<Vec<String>>().join(",");`, but ok.

File: tests/integration/utils/chroma_log_collector.py (hunk 4 above)
Comment (johnsonw) [reply tests/integration/utils/chroma_log_collector.py]: Looks like it got formatted.

File: iml-services/iml-action-runner/src/receiver.rs [hunk 6]
```diff
@@ -26,6 +26,47 @@ fn terminate_session(fqdn: &Fqdn, sessions: &mut Sessions, session_to_rpcs: &mut
...
+    };
+
+    if let Some(old_id) = maybe_old_id {
+        if let Some(xs) = { rpcs.lock().await.remove(&old_id) } {
+            for action_in_flight in xs.values() {
+                let session_id = session_id.clone();
+                let fqdn = fqdn.clone();
+                let body = action_in_flight.action.clone();
+
+                let msg = create_data_message(session_id, fqdn, body);
```
[the remark this answers is not in the document]
Comment (jgrund) [reply iml-services/iml-action-runner/src/receiver.rs]: Not without updating the struct these are used in to take lifetimes.
